ml-agents/mlagents/trainers/components/bc/module_torch.py

```diff
@@ -0,0 +1,192 @@
+from typing import Dict, Any
+import numpy as np
+import torch
+
+from mlagents.trainers.policy.torch_policy import TorchPolicy
+from mlagents.trainers.demo_loader import demo_to_buffer
+from mlagents.trainers.settings import BehavioralCloningSettings
+from mlagents.trainers.torch.utils import ModelUtils
+
+
+class TorchBCModule:
+    def __init__(
+        self,
+        policy: TorchPolicy,
+        settings: BehavioralCloningSettings,
+        policy_learning_rate: float,
+        default_batch_size: int,
+        default_num_epoch: int,
+    ):
+        """
+        A BC trainer that can be used inline with RL.
+        :param policy: The policy of the learning model
+        :param policy_learning_rate: The initial Learning Rate of the policy. Used to set an appropriate learning rate
+            for the pretrainer.
+        :param default_batch_size: The default batch size to use if batch_size isn't provided.
+        :param default_num_epoch: The default num_epoch to use if num_epoch isn't provided.
+        :param strength: The proportion of learning rate used to update through BC.
+        :param steps: The number of steps to anneal BC training over. 0 for continuous training.
+        :param demo_path: The path to the demonstration file.
+        :param batch_size: The batch size to use during BC training.
+        :param num_epoch: Number of epochs to train for during each update.
+        :param samples_per_update: Maximum number of samples to train on during each BC update.
+        """
+        self.policy = policy
+        self.current_lr = policy_learning_rate * settings.strength
+        params = list(self.policy.actor_critic.parameters())
```

Comment:
Will policy always have a public actor_critic? I wonder if we should have a `parameters` method on `TorchPolicy` to make this more easy to understand.